File: nestjs/src/business-rules/customers/handlers/create-customer.handler.ts
import { ICustomerResponse } from './../responses/icustomer.response';
import { Inject, Injectable } from '@nestjs/common';
import { ICreateCustomerRequest } from '../requests/icreate-customer.request';
import {
  ICustomerRepository,
  ICustomerRepositoryName,
} from '@domain/repositories';
import { ICreateCustomerHandler } from './icreate-customer.handler';

@Injectable()
export class CreateCustomerHandler implements ICreateCustomerHandler {
  constructor(
    @Inject(ICustomerRepositoryName) private repository: ICustomerRepository,
  ) {}

  async execute(request: ICreateCustomerRequest): Promise<ICustomerResponse> {
    const customer = await this.repository.create({
      name: request.name,
      email: request.email,
      birthDate: request.birthDate,
    });

    const response: ICustomerResponse = {
      payload: {
        id: customer.id,
        name: customer.name,
        email: customer.email,
        birthDate: customer.birthDate,
      },
    };

    return response;
  }
}

File: dotnet/BusinessRules/Customers/Handlers/CreateCustomerHandler.cs
using dotnet.BusinessRules.Customers.Requests;
using dotnet.BusinessRules.Customers.Responses;
using dotnet.Domain.Models;
using dotnet.Domain.Repositories;

namespace dotnet.BusinessRules.Customers.Handlers
{
    public class CreateCustomerHandler : ICreateCustomerHandler
    {
        private readonly ICustomerRepository _customerRepository;

        public CreateCustomerHandler(
            ICustomerRepository customerRepository
        )
        {
            _customerRepository = customerRepository;
        }

        public CustomerResponse ExecuteAsync(CreateCustomerRequest request)
        {
            var customer = _customerRepository.Create(new Customer
            {
                Name = request.Email,
                Email = request.Email,
                BirthDate = request.BirthDate
            });

            return new CustomerResponse
            {
                Payload = new CustomerItemResponse
                {
                    Id = customer.Id,
                    Name = customer.Name,
                    Email = customer.Email,
                    BirthDate = customer.BirthDate
                }
            };
        }
    }
}